nodesigner.go

```diff
@@ -44,6 +44,33 @@ func (n *nodeSigner) SignMessage(pubKey *btcec.PublicKey,
 	return sign, nil
 }
 
+// SignCompact signs a double-sha256 digest of the passed msg under the
+// resident node's private key. If the target public key is _not_ the node's
+// private key, then an error will be returned. The returned signature is a
+// pubkey-recoverable signature.
+func (n *nodeSigner) SignCompact(pubKey *btcec.PublicKey,
+	msg []byte) ([]byte, error) {
+
+	// If this isn't our identity public key, then we'll exit early with an
+	// error as we can't sign with this key.
+	if !pubKey.IsEqual(n.privKey.PubKey()) {
+		return nil, fmt.Errorf("unknown public key")
+	}
+
+	// Otherwise, we'll sign the dsha256 of the target message.
```

Comment:
In line with the contribution guidelines of this project, the code fragment under this line should be modified to have some [breathing room](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeSpacing).